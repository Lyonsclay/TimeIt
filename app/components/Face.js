import React from 'react';
import {
  View,
  Animated,
  Easing
} from 'react-native';
import Svg, {
  Circle, G
} from 'react-native-svg';

const Face = (props) => {
  const {
    width,
    height,
    radius,
    strokeWidth
  } = props;
  const cx = radius + strokeWidth / 2;
  const cy = cx;
  const faceStyle = {
    position: 'absolute',
  };
  const pulse = new Animated.Value(0)

  const visualAlarm = () => {
    Animated.sequence([
      Animated.timing(
        pulse,
        {
          toValue: 1,
          duration: 250,
          easing: Easing.linear
        }),
      Animated.timing(
        pulse,
        {
          toValue: 0,
          duration: 250,
          easing: Easing.ease
        })
    ]
    ).start(visualAlarm)
  }

  if (props.clock.alarm && (props.app.timerView[0] === 'PAUSE')) {visualAlarm()
  }

  return (
    <Svg
      style={faceStyle}
      width={width}
      height={height}
    >
      <Circle
        cx={cx}
        cy={cy}
        r={radius}
        fillOpacity='1'
        fill='orange'
        stroke='#728c00'
        strokeWidth={strokeWidth}
      />
      <Animated.View style={{opacity: pulse}}>
        <Svg
          style={faceStyle}
          width={width}
          height={height}
        >
          <Circle
            cx={cx}
            cy={cy}
            r={radius}
            fill='red'
            fillOpacity='1'
            stroke='#728c00'
            strokeWidth={strokeWidth}
          />
        </Svg>
      </Animated.View>
    </Svg>
  );
};

export default Face;
//
