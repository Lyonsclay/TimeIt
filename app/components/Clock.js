import React, {
  Component
} from 'react'
import {
  connect
} from 'react-redux'
import {
  showTimeSet,
  hideTimeSet
} from '../actions'
import {
  StyleSheet,
  View,
  TouchableOpacity
} from 'react-native';
import Svg, {
  Circle,
  Line
} from 'react-native-svg';
import Face from './Face';
import Hand from './Hand';

const Clock = (props) => {
  const {
    clock,
    screen
  } = props
  const diameter = clock.diameter;
  const radius = diameter / 2;
  const {
    xMid,
    yMid
  } = screen;
  const top = yMid - radius;
  const left = xMid - radius;
  const width = diameter + clock.strokeWidth;
  const height = width;
  const clockStyle = {}

  const _showTimeSet = () => {
    alert(JSON.stringify(props))
    props.showTimeSet()
  }

  return (
    <TouchableOpacity onPress={props.showTimeSet}>
      <View style={styles.container}>
        <Face
          strokeWidth={clock.strokeWidth}
          width={width}
          height={height}
          radius={radius}
        />
        <View style={clockStyle} >
          <Hand
            style={clockStyle}
            strokeWidth={clock.strokeWidth}
            width={width}
            height={height}
            radius={radius}
          />
        </View>
      </View>
    </TouchableOpacity>
  )
}

const styles = StyleSheet.create({
  container: {
    flex: 0
  }
});

/* 
 * const mapStateToProps = (state) => {
 *   return state
 * }
 * const mapDispatchToProps = (dispatch) => {
 *   return {
 *     showTimeSet: () => dispatch(showTimeSet()),
 *     hideTimeSet: () => dispatch(hideTimeSet())
 *   }
 * }*/

export default Clock;
