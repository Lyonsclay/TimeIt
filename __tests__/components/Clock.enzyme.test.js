import { View, Text } from 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import Clock from '../../app/components/Clock';
import Face from '../../app/components/Face';
import Svg, { Line, Circle } from 'react-native-svg';

class Nope extends React.Component {
  render() {
    return (
      <Face
        strokeWidth={2.5}
        width={200}
        height={200}
        radius={99}
      />
    );
  }
}

describe('Face', () => {
  const props = {
    strokeWidth: 2.5,
    width: 101.5,
    height: 101.5,
    radius: 49.5
  };

  it('renders', () => {
    let face = shallow(<View><Face {...props} /></View>);
    expect(face).toBeDefined();
    expect(face.length).toEqual(1);
    expect(face).toMatchSnapshot();
    expect(face.children().props().radius).toEqual(49.5);
  });
});

describe('Clock', () => {
  const initialState = {
    screen: {
      width: 0,
      height: 0,
      xMid: 0,
      yMid: 0
    },
    clock: {
      diameter: 99,
      strokeWidth: 2.5,
      duration: 0.0,
      moving: false
    }
  };
  const props = {
    strokeWidth: 2.5,
    width: 101.5,
    height: 101.5,
    radius: 49.5
  };

  it('renders', () => {
    const clock = shallow(<Clock {...initialState} />);
    const face = () => (<Face {...props} />);
    expect(clock.contains(<Face {...props} />)).toBe(true);
    expect(clock.contains(face())).toBe(true);
  });
});

describe('RN View and Text', () => {
  const Funny = () => {
    return (
      <View><Text>Funny</Text></View>
    );
  };

  it('contains Text', () => {
    const view = shallow(<Funny />);
    expect(view.contains(<Text>Funny</Text>)).toBe(true);
  });
});
