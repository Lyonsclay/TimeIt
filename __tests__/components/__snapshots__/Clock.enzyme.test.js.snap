exports[`Face renders 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function buildPredicate],
    "childrenOfNode": [Function childrenOfNode],
    "findWhereUnwrapped": [Function findWhereUnwrapped]
  },
  "length": 1,
  "node": <View>
    <[object Object]
        height={101.5}
        radius={49.5}
        strokeWidth={2.5}
        width={101.5} />
</View>,
  "nodes": Array [
    <View>
      <[object Object]
            height={101.5}
            radius={49.5}
            strokeWidth={2.5}
            width={101.5} />
</View>
  ],
  "options": Object {},
  "renderer":  {
    "_instance":  {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <function Component(){_classCallCheck(this,Component);return _possibleConstructorReturn(this,(Component.__proto__||Object.getPrototypeOf(Component)).apply(this,arguments));}>
        <[object Object]
                height={101.5}
                radius={49.5}
                strokeWidth={2.5}
                width={101.5} />
</function Component(){_classCallCheck(this,Component);return _possibleConstructorReturn(this,(Component.__proto__||Object.getPrototypeOf(Component)).apply(this,arguments));}>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Component {
        "_reactInternalInstance": [Circular],
        "blur": [Function bound ],
        "context": Object {},
        "focus": [Function bound ],
        "measure": [Function bound ],
        "measureInWindow": [Function bound ],
        "measureLayout": [Function bound ],
        "props": Object {
          "children": <[object Object]
            height={101.5}
            radius={49.5}
            strokeWidth={2.5}
            width={101.5} />
        },
        "refs": Object {},
        "setNativeProps": [Function bound ],
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function anonymous],
          "enqueueCallbackInternal": [Function anonymous],
          "enqueueElementInternal": [Function anonymous],
          "enqueueForceUpdate": [Function anonymous],
          "enqueueReplaceState": [Function anonymous],
          "enqueueSetState": [Function anonymous],
          "isMounted": [Function anonymous],
          "validateCallback": [Function anonymous]
        }
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": Object {
        "_currentElement": <View>
          <[object Object]
                    height={101.5}
                    radius={49.5}
                    strokeWidth={2.5}
                    width={101.5} />
</View>,
        "_debugID": 2,
        "_renderedOutput": <View>
          <[object Object]
                    height={101.5}
                    radius={49.5}
                    strokeWidth={2.5}
                    width={101.5} />
</View>
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false
    },
    "getRenderOutput": [Function getRenderOutput],
    "render": [Function render]
  },
  "root": [Circular],
  "unrendered": <function Component(){_classCallCheck(this,Component);return _possibleConstructorReturn(this,(Component.__proto__||Object.getPrototypeOf(Component)).apply(this,arguments));}>
    <[object Object]
        height={101.5}
        radius={49.5}
        strokeWidth={2.5}
        width={101.5} />
</function Component(){_classCallCheck(this,Component);return _possibleConstructorReturn(this,(Component.__proto__||Object.getPrototypeOf(Component)).apply(this,arguments));}>
}
`;
